include/envoy/stats/stats.h

```diff
@@ -156,5 +156,25 @@ class Gauge : public Metric {
 
 using GaugeSharedPtr = RefcountPtr<Gauge>;
 
+/**
+ * A string, possibly non-ASCII.
+ */
+class TextReadout : public virtual Metric {
+public:
+  // Text readout type is used internally to disambiguate isolated store
+  // constructors. In the future we can extend it to specify text encoding or
+  // some such.
+  enum class Type {
+    Default, // No particular meaning.
+  };
+
+  virtual ~TextReadout() {}
+
+  virtual void set(std::string&& value) PURE;
```

Comment:
Needs doc, especially since it's got a potentially unexpected (though justified) behavior. In fact I might even call this 'move' rather than 'set'?

doc also for the accessor.

Reply:
Done.